### PR TITLE
Add missing AT lines to allow registering custom game rule value types

### PR DESCRIPTION
**Pull request by ApexModder**
This PR adds the missing lines to the access transformer file to allow modders to register custom game rule value types, which are required in the game rules registration method `GameRules.register`.

These methods are `GameRules.BooleanValue.create` & `GameRules.IntegerValue.create`.

Forge previously only added an AT line for the game rules register method, but without AT'ing the value type methods modders could not register new game rules, as the register method takes in a value type.

While modders could easily AT these methods, forge should really be AT'ing them since they are already AT'ing the register method.

---

This PR also introduces a basic test to validate that custom game rules can be registered.

<details>

This test registers 2 new game rules
- `custom_game_rule_test:custom_boolean_game_rule`
- `custom_game_rule_test:custom_integer_game_rule`

To validate that these game rules were registered correctly
- Create a new world and ensure they show up under the `Edit Game Rules` screen (`Create new world > More (tab) > Game Rules > Misc`)
- Try and change their values in game using the following commands
	_These game rules should come up as valid auto-completion values if registered._
- `/gamerule custom_game_rule_test:custom_boolean_game_rule <true|false>`
- `/gamerule custom_game_rule_test:custom_integer_game_rule <some integer>`

</details>

### CODE REVIEW

File: src/main/resources/META-INF/accesstransformer.cfg [hunk 1]
```diff
@@ -456,4 +460,4 @@ public net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType <
 protected net.minecraft.world.level.portal.PortalForcer f_77648_ # level
 public net.minecraft.world.level.storage.LevelResource <init>(Ljava/lang/String;)V # constructor
 private-f net.minecraft.world.level.storage.loot.LootPool f_79028_ # rolls
-private-f net.minecraft.world.level.storage.loot.LootPool f_79029_ # bonusRolls
\ No newline at end of file
+private-f net.minecraft.world.level.storage.loot.LootPool f_79029_ # bonusRolls
```
Comment (ChampionAsh5357) [inline src/main/resources/META-INF/accesstransformer.cfg]: Remove unnecessary change

Comment (ApexModder) [reply src/main/resources/META-INF/accesstransformer.cfg]: My bad I had a setting enabled that told idea to ensure every file ended with a new line, disabled that, and reverted the change (removed the blank new line).

**Review by ChampionAsh5357 — APPROVED** [review]
